packages/client/src/components/app/Modal.svelte

```diff
@@ -0,0 +1,139 @@
+<script>
+  import { getContext } from "svelte"
+  import { Modal, Icon } from "@budibase/bbui"
+
+  const component = getContext("component")
+  const { styleable, modalStore, builderStore, dndIsDragging } =
+    getContext("sdk")
+
+  export let onClose
+  export let ignoreClicksOutside
+  export let size
+  let modal
+
+  // Open modal automatically in builder
+  $: {
+    if ($builderStore.inBuilder) {
+      if (
+        $component.inSelectedPath &&
+        $modalStore.contentId !== $component.id
+      ) {
+        modalStore.actions.open($component.id)
+      } else if (
+        !$component.inSelectedPath &&
+        $modalStore.contentId === $component.id &&
+        !$dndIsDragging
+      ) {
+        modalStore.actions.close()
+      }
+    }
+  }
+
+  $: open = $modalStore.contentId === $component.id
+
+  const handleModalClose = async () => {
+    if (onClose) {
+      await onClose()
+    }
+    modalStore.actions.close()
+  }
+
+  const handleOpen = (open, modal) => {
+    if (!modal) return
+
+    if (open) {
+      modal.show()
+    } else {
+      modal.hide()
+    }
+  }
+
+  $: handleOpen(open, modal)
+</script>
+
+<Modal
+  customMargin="20px"
+  on:cancel={handleModalClose}
+  bind:this={modal}
+  disableCancel={$builderStore.inBuilder}
+  zIndex={2}
```

Comment:
Purely out of curiosity - what's the z index of 2 for?

Reply:
This puts it above the app header which seems to be `z-index: 1` but below the selection indicators.

the default `999` value the modal has for this makes it so that the selection/highlight indicators for the modal appear under the translucent overlay.